samtranslator/metrics/metrics.py

```diff
@@ -0,0 +1,156 @@
+import logging
+
+LOG = logging.getLogger(__name__)
+
+
+class MetricsPublisher:
+    """Interface for all MetricPublishers"""
+
+    def __init__(self):
+        pass
+
+    def publish(self, namespace, metrics):
+        raise NotImplementedError
+
+
+class CWMetricsPublisher(MetricsPublisher):
+    BATCH_SIZE = 20
+
+    def __init__(self, cloudwatch_client):
+        """
+        Constructor
+
+        :param cloudwatch_client: cloudwatch client required to publish metrics to cloudwatch
+        """
+        MetricsPublisher.__init__(self)
+        self.cloudwatch_client = cloudwatch_client
+
+    def publish(self, namespace, metrics):
+        """
+        Method to publish all metrics to Cloudwatch.
+
+        :param namespace: namespace applied to all metrics published.
+        :param metrics: list of metrics to be published
+        """
+        batch = []
+        for metric in metrics:
+            batch.append(metric)
+            # Cloudwatch recommends not to send more than 20 metrics at a time
+            if len(batch) == self.BATCH_SIZE:
+                self._flush_metrics(namespace, batch)
+                batch = []
+        self._flush_metrics(namespace, batch)
+
+    def _flush_metrics(self, namespace, metrics):
+        """
+        Internal method to publish all provided metrics to cloudwatch, please make sure that array size of metrics is <= 20.
+        """
+        metric_data = list(map(lambda m: m.get_metric_data(), metrics))
+        try:
+            self.cloudwatch_client.put_metric_data(Namespace=namespace, MetricData=metric_data)
+        except Exception as e:
+            LOG.error("Failed to report {} metrics".format(len(metric_data)))
+            LOG.exception(e)
```

Comment:
nit: I think there is a parameter in the logging framework to pass the exception rather than doing it in 2 lines

Reply:
Fixed!!